components/engine_rocks/src/raft_engine.rs

```diff
@@ -0,0 +1,261 @@
+    }
+
+    fn get_entry(&self, raft_group_id: u64, index: u64) -> Result<Option<Entry>> {
+        let key = keys::raft_log_key(raft_group_id, index);
+        let entry = box_try!(self.get_msg_cf(CF_DEFAULT, &key));
+        Ok(entry)
+    }
+
+    fn fetch_entries_to(
+        &self,
+        region_id: u64,
+        low: u64,
+        high: u64,
+        max_size: Option<usize>,
+        buf: &mut Vec<Entry>,
+    ) -> Result<usize> {
+        let (max_size, mut total_size, mut count) = (max_size.unwrap_or(usize::MAX), 0, 0);
+
+        if high - low <= RAFT_LOG_MULTI_GET_CNT {
+            // If election happens in inactive regions, they will just try to fetch one empty log.
+            for i in low..high {
+                if total_size > 0 && total_size >= max_size {
+                    break;
+                }
+                let key = keys::raft_log_key(region_id, i);
+                match self.get_value(&key) {
+                    Ok(None) => return Err(Error::Storage(StorageError::Unavailable)),
+                    Ok(Some(v)) => {
+                        let mut entry = Entry::default();
+                        entry.merge_from_bytes(&v)?;
+                        assert_eq!(entry.get_index(), i);
+                        buf.push(entry);
+                        total_size += v.len();
+                        count += 1;
+                    }
+                    Err(e) => return Err(box_err!(e)),
+                }
+            }
+            return Ok(count);
+        }
+
+        let (mut check_compacted, mut next_index) = (true, low);
+        let start_key = keys::raft_log_key(region_id, low);
+        let end_key = keys::raft_log_key(region_id, high);
+        box_try!(self.scan(
+            &start_key,
+            &end_key,
+            true, // fill_cache
+            |_, value| {
+                let mut entry = Entry::default();
+                entry.merge_from_bytes(value)?;
+
+                if check_compacted {
+                    if entry.get_index() != low {
+                        // May meet gap or has been compacted.
+                        return Ok(false);
+                    }
+                    check_compacted = false;
+                } else {
+                    assert_eq!(entry.get_index(), next_index);
+                }
+                next_index += 1;
+
+                buf.push(entry);
+                total_size += value.len();
+                count += 1;
+                Ok(total_size < max_size)
+            },
+        ));
+
+        // If we get the correct number of entries, returns.
+        // Or the total size almost exceeds max_size, returns.
+        if count == (high - low) as usize || total_size >= max_size {
+            return Ok(count);
+        }
+
+        // Here means we don't fetch enough entries.
+        Err(Error::Storage(StorageError::Unavailable))
+    }
+
+    fn consume(&self, batch: &mut Self::LogBatch, sync_log: bool) -> Result<usize> {
```

Comment:
The name of this method 'consume' looks strange to me.... 
Why not use 'write' instead?